llama-index-integrations/vector_stores/llama-index-vector-stores-qdrant/llama_index/vector_stores/qdrant/utils.py

```diff
@@ -75,10 +75,22 @@ def fastembed_sparse_encoder(
     except ImportError as e:
         raise ImportError(
             "Could not import FastEmbed. "
-            "Please install it with `pip install fastembed`"
+            "Please install it with `pip install fastembed` or "
+            "`pip install fastembed-gpu` for GPU support.\n"
+            "See `https://qdrant.github.io/fastembed/examples/FastEmbed_GPU/` for more details on GPU support."
         ) from e
 
-    model = SparseTextEmbedding(model_name, cache_dir=cache_dir, threads=threads)
+    # prioritize GPU over CPU
+    try:
+        model = SparseTextEmbedding(
+            model_name,
+            cache_dir=cache_dir,
+            threads=threads,
+            providers=["CUDAExecutionProvider"],
```

Comment:
I would recommend to parameterize this. If someone intends to use it, they can install `fastembed-gpu`. `fastembed` otherwise.

Reply:
Yea it's just a tad annoying to pass this deep 😅 The `init` is kind of already overflowing with options. Let's keep it for now and consider that future optimization. The user can always pass in a custom sparse function instead

Reply:
I am not so sure about assuming providers. Like CUDA in this case.

Reply:
Understood. I guess overriding the sparse function with custom requirements would be better anyways.